src/core/ext/filters/client_channel/lb_policy/round_robin/round_robin.c

```diff
@@ -74,6 +74,9 @@ typedef struct round_robin_lb_policy {
   bool started_picking;
   /** are we shutting down? */
   bool shutdown;
+  /** has the policy gotten into the GRPC_CHANNEL_SHUTDOWN? No picks can be
+   * service after this point, the policy will never transition out. */
+  bool in_connectivity_shutdown;
```

Comment:
I have the same mixed feelings.  If you want to revert this, that's fine with me -- your call.